flybywiresim/a32nx
# fix(atsu): Hoppie disconnect in mid-flight and ingame ATC

Fixes reports in Discord

## Summary of Changes
Some reports in discord indicated that the sim var ATC FLIGHT NUMBER is sometimes changed in the simulator without notifying the the MCDU. Additionally was reported that the callsign is ignored by the ingame ATC on the ground.

Could not reproduce the ATC FLIGHT NUMBER changes, but decoupled this sm var from the callsign in hoppie or telex.
The flight number "master" is now the MCDU and the ATC FLIGHT NUMBER is ignored by the ATSU.

Both issues are fixed with this PR.

## Screenshots (if necessary)



## References





## Additional context



Discord username (if different from GitHub): Sven [de en]

## Testing instructions

- Fly with ingame ATC and check if the entered flight number (in the MSFS menu) is recognized on ground and in air

- Check ATSU related pages (ATC Notification, AOC DEPART REQ)
  - Enter on INIT A a flight number and check if the pages are updated

- Check if Hoppie connect is validated correct
  - Connect to Hoppie without an entered flight number -> expect no error
  - Connect to Hoppie with an available flight number -> expect no error
  - Connect to Hoppie with an callsign in use -> expect a callsign in use error

- Check connection to Hoppie
  - Test if messages are sent and received under the following situation
    - validate that the correct callsign is used for the Hoppie connection (http://www.hoppie.nl/acars/system/online.html)
    - set MCDU -> connect to hoppie
    - connect to hoppie -> set MCDU

- Check MCDU FLT NO updates with Hoppie connection
  - Available callsign -> no error
  - Used callsign -> FLT/NO ALREADY IN USE

- Disconnection from Hoppie with active ATC connection
  - Logon to dummy ATC
  - check Comm Status if CURRENT ATC is correct
  - send command from ATC to A32NX and keep message on DCDU
  - Hoppie Disconnect in EFB
    - Cleaned up DCDU
    - Cleaned up Comm Status
  - The same should be valid for ongoing logon


## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
